util/encoding/encoding_test.go

```diff
@@ -362,6 +362,24 @@ func TestDecodeInvalidUvarint(t *testing.T) {
 	}
 }
 
+// TestDecodeInvalidVarint tests that invalid uvarint encodings panic.
+func TestDecodeInvalidVarint(t *testing.T) {
+	testCases := [][]byte{
+		// overflows int64.
+		{0x10, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff},
+	}
+	for _, c := range testCases {
+		func() {
+			defer func() {
+				if r := recover(); r == nil {
+					t.Error("expected panic")
+				}
```

Comment:
This is a good idea. I have the feeling that doing this in tests would encourage descriptive errors.

Done.